### PR TITLE
Azure Resource Metrics - Apply doc template and add generic setup section

### PR DESCRIPTION
**Pull request by alaudazzi**
This PR:
- applies the [doc template](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md) to [Resource metrics](https://docs.elastic.co/integrations/azure_metrics)
- integrates a more [generic setup section](https://docs.google.com/document/d/1q30RS_qhTuDwn5OVstXC5EofHMhKvNXktxY7gEKWExI/edit) similar to the one in [Azure Billing metrics](https://docs.elastic.co/integrations/azure_billing)
- harmonizes the text according to the new structure

Relates [#123](https://github.com/elastic/integration-docs/issues/310)

### CODE REVIEW

**Comment by alaudazzi** [comment]
@zmoog 
Here is a [doc preview](https://github.com/elastic/integrations/blob/ee4529c7f7bab62349b77f21558b8474c3203837/packages/azure_metrics/docs/README.md).

File: packages/azure_metrics/_dev/build/docs/README.md [hunk 1]
```diff
@@ -99,9 +68,134 @@ so the `period` for `database_account` should be `300s` or multiples of `300s`.
...
+Take note of the content in the **Value** column in the **Client secrets** table, which you will use later when specifying a **Client Secret** in the integration settings. **This secret value is never displayed again after you leave this page.** Record the secret's value in a safe place.
+
+**Step 3: Assign role**
+
+
+1. In the [Azure Portal](https://portal.azure.com/), search for and select **Subscriptions**.
+1. Select the subscription to assign the application.
+1. Select **Access control (IAM)**.
+1. Select **Add** > **Add role assignment** to open the _Add role assignment page_.
+1. In the **Role** tab, search and select the role **Billing Reader**.
```
Comment (zmoog) [inline packages/azure_metrics/_dev/build/docs/README.md:140]:
```suggestion
1. In the **Role** tab, search and select the role **Monitoring Reader**.
```

File: packages/azure_metrics/_dev/build/docs/README.md [hunk 2]
```diff
@@ -99,9 +68,134 @@ so the `period` for `database_account` should be `300s` or multiples of `300s`.
...
+
+Your App Registration is now ready for the Elastic Agent.
+
+### Specify the integration settings in Kibana
+
+Add the Azure Resource Metrics integration in Kibana and specify settings.
+
+If you're new to integrations, you can find step-by-step instructions on how to set up an integration in the {{ url "getting-started-observability" "Getting started" }} guide.
+
+The settings' main section contains all the options needed to access the Azure APIs and collect the billing data. You will now use all the values from [App registration](#app-registration) including:
```
Comment (zmoog) [inline packages/azure_metrics/_dev/build/docs/README.md:160]:
```suggestion
The settings' main section contains all the options needed to access the Azure APIs and collect the monitoring data. You will now use all the values from [App registration](#app-registration) including:
```

File: packages/azure_metrics/_dev/build/docs/README.md [hunk 3]
```diff
@@ -99,9 +68,134 @@ so the `period` for `database_account` should be `300s` or multiples of `300s`.
...
+
+`Client ID` _string_
+: The unique identifier of the App Registration (sometimes referred to as Application ID).
+
+`Client Secret` _string_
+: The client secret for authentication.
+
+`Subscription ID` _string_
+: The unique identifier for the Azure subscription. You can provide just one subscription ID. The Agent uses this ID to access Azure APIs. The Agent also uses this ID as the default scope for billing information: see the "Scope" section for more details about how to collect data for more than one subscription.
```
Comment (zmoog) [inline packages/azure_metrics/_dev/build/docs/README.md:169]: This does not apply to non-billing metrics:

```suggestion
: The unique identifier for the Azure subscription. You can provide just one subscription ID. The Agent uses this ID to access Azure APIs. 
```

File: packages/azure_metrics/_dev/build/docs/README.md [hunk 4]
```diff
@@ -2,56 +2,25 @@
...
+- `compute_vm`
+- `compute_vm_scaleset`
+- `storage_account`
+- `container_instance`
+- `container_registry`
+- `container_service`
+- `database_account`
 
- `monitor`
+`monitor`
```
Comment (zmoog) [inline packages/azure_metrics/_dev/build/docs/README.md:23]: After reading the [preview](https://github.com/elastic/integrations/blob/ee4529c7f7bab62349b77f21558b8474c3203837/packages/azure_metrics/docs/README.md), I have the feeling that we should move this section with the metricset description and tables (generaged by `{{fields "monitor"}}`) at the bottom of the doc in a "references" section.

The `{{fields "monitor"}}` expands in a large table, and IMO it's easy to miss the content after the tables.

What about adding something in the lines of "see references section for all the gory details of these metricsets" right after the metricset list?

Comment (alaudazzi) [reply packages/azure_metrics/_dev/build/docs/README.md:23]: Yes 👍  that's what I also suggested [here](https://github.com/elastic/integrations/pull/9065#issuecomment-1931510851). If the test with collapse/expand feature doesn't work, we'll put the tables at the end of the page and reference them where needed.

**Comment by alaudazzi** [comment]
@zmoog 
Consider some further optimization on this page:

The first block of text of [Integration specific configuration notes](https://docs.elastic.co/integrations/azure_metrics/container_service#integration-specific-configuration-notes) and the entire section of [Additional notes about metrics and costs](https://docs.elastic.co/integrations/azure_metrics/container_service#additional-notes-about-metrics-and-costs) can be combined into a single and more linear section called **Authentication and costs**.

For example:
____________
image

**Comment by zmoog** [comment]
> The first block of text of [Integration specific configuration notes](https://docs.elastic.co/integrations/azure_metrics/container_service#integration-specific-configuration-notes) and the entire section of [Additional notes about metrics and costs](https://docs.elastic.co/integrations/azure_metrics/container_service#additional-notes-about-metrics-and-costs) can be combined into a single and more linear section called **Authentication and costs**.

Yep, it makes senso combine these two sections into one.

Two notes on the content:

- In all our Azure-related docs, we always use the term "Azure AD" (or "Azure Active Directory"). It seems that Microsoft renamed Azure AD in Microsoft Entra ID. Updating these names in all docs will probably require a dedicated PR, so I don't know if we can start updating the portions we change in this one.
- In these section, we use the term "service principal". IIRC, this is a different name for the "app registration" concept. We need to double-check if "service principal" and "app registration" are the same thing or not and decide how to deal with them.

**Comment by zmoog** [comment]
If we want to continue using the ASCII art for diagrams, I can share the text for [azure_resource_metrics_req.png](https://github.com/elastic/integrations/pull/9065/files#diff-9dfcfbf9d10f7a7195f3165778b8b9752fca0e4b252f77b356e7d00588baaa64).

**Comment by zmoog** [comment]
> In these section, we use the term "service principal". IIRC, this is a different name for the "app registration" concept. We need to double-check if "service principal" and "app registration" are the same thing or not and decide how to deal with them.

The name "service principal" is commonly used to represent the identity of an application for authentication purposes. Other cloud providers, for example GCP, use the term service principal for this concept. 

On Azure, they seem to use "app registration" on the Portal, and "service principal" on the CLI and API:

- https://learn.microsoft.com/en-us/cli/azure/azure-cli-sp-tutorial-1?tabs=bash

On the Azure CLI, if I list the service principals:

```bash
$ az ad sp list | jq -r '.[] | [.appDisplayName] | @tsv' | grep branca

mbranca-app-registration-or-service-principal
mbranca-elastic-agent
```

I get the same result if I open the app registrations section on the Portal:

CleanShot 2024-02-12 at 17 46 55@2x

**Comment by alaudazzi** [comment]
@zmoog 

> On Azure, they seem to use "app registration" on the Portal, and "service principal" on the CLI and API:

So, the concept is the same, the only difference is that `app registration` is on the UI and `service principal` is on the CLI?

> Microsoft renamed Azure AD in Microsoft Entra ID

We fix the occurrences in this PR and handle the remaining ones in a separate PR

**Comment by zmoog** [comment]
> The name "service principal" is commonly used to represent the identity of an application for authentication purposes. Other cloud providers, for example GCP, use the term service principal for this concept.

It turns out that on Azure "app registration" and "service principal" are related but NOT the same.

Quoting [Create a Microsoft Entra application and service principal that can access resources](https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal) from the Azure docs:

> When you register a new application in Microsoft Entra ID, a service principal is automatically created for the app registration. The service principal is the app's identity in the Microsoft Entra tenant.

So, "app registration" and "service principal" are different entities. Azure creates the related service principal for you if you create an app registration. And it works both ways!

So users have the option of creating:

 - (A) an app registration on the Azure Portal (probably easier to do manually), and get the service principal for free.
 - (B) a service registration using the CLI or the API (probably more accessible to automate), and get an app registration for free.

**Comment by zmoog** [comment]
I would use the app registration in our docs to resolve the "app registration" vs. "service principal" dilemma. The App registration is the most visible term in Azure Portal, the UI we primarily use in our docs.

Then, if we link to Azure docs that explain how to create a service principal using the CLI (bash or PowerShell), in that case, we can mention that creating a service principal will also create a linked app registration that will be visible on the Portal.

@alaudazzi, WDYT?

**Comment by alaudazzi** [comment]
> I would use the app registration in our docs to resolve the "app registration" vs. "service principal" dilemma. The App registration is the most visible term in Azure Portal, the UI we primarily use in our docs.

makes sense

> Then, if we link to Azure docs that explain how to create a service principal using the CLI (bash or PowerShell), in that case, we can mention that creating a service principal will also create a linked app registration that will be visible on the Portal.

I added a note in the **Authentication and costs** section.

The PR is ready for you to review. I guess we integrated all the comments raised so far.

**Review by zmoog — APPROVED** [review]
LGTM!

nit: I think we can remove

**Review by zmoog — APPROVED** [review]
LGTM!

Nit: I think we can remove the .png file since we opted to use the ASCII version of the diagram.

**Pull request merged**